raven/src/main/java/org/opencadc/raven/ProtocolsGenerator.java

```diff
@@ -366,6 +361,22 @@ List<Protocol> doPullFrom(URI artifactURI, Transfer transfer, String authToken)
                 }
             }
         }
+        if ((storageResolvers != null) && storageResolvers.containsKey(artifactURI.getScheme())) {
+            URL externalURL = storageResolvers.get(artifactURI.getScheme()).toURL(artifactURI);
+            for (Protocol proto : transfer.getProtocols()) {
+                proto.setEndpoint(externalURL.toString());
+                protos.add(proto);
+                log.debug("added external " + proto);
+            }
+        }
```

Comment:
I'm likely not understanding how StorageResolvers and Protocols work, but isn't the StorageResolver endpoint overwriting the StorageSite endpoint set above?

Reply:
Also a good point. These are the requested protocols and I should have cloned instead of reusing them as it's the case with the StorageSite end points above. Fixed now.